### ISSUE TITLE
github build failing due to git bad object

### ISSUE THREAD

**Issue by stitakis** [opendevstack/ods-provisioning-app#595]
The build implemented with github actions is failing randomly at stage ` Build docker image`.
The error displayed points to a git bad object:
```
-> Run COMMIT_AUTHOR=$(git --no-pager show -s --format='%an (%ae)' $GITHUB_SHA)
fatal: bad object 94a87e4f6ad38d7354fa3c49769192429b4344da
```

**Comment by stitakis** [comment]
I added some logging to the github actions and run the build a couple of times.

It looks like the variable `GITHUB_SHA` sometimes has a different value that the branch commit sha like in the following example:
```
Run COMMIT_SHA=$(git show --format="%H" --no-patch)
  COMMIT_SHA=$(git show --format="%H" --no-patch)
  COMMIT_AUTHOR=$(git --no-pager show -s --format='%an (%ae)' $COMMIT_SHA)
  COMMIT_MESSAGE=$(git log -1 --pretty=%B $COMMIT_SHA)
  COMMIT_TIME=$(git show -s --format=%ci $COMMIT_SHA)
  BUILD_TIME=$(date -u "+%Y-%m-%d %H:%M:%S %z")
  echo "GITHUB_REF=$GITHUB_REF" \
  echo "COMMIT_AUTHOR=$COMMIT_AUTHOR" \
  echo "COMMIT_MESSAGE=$COMMIT_MESSAGE" \
  echo "COMMIT_SHA=$COMMIT_SHA" \
  echo "GITHUB_SHA=$GITHUB_SHA" \
  echo "COMMIT_TIME=$COMMIT_TIME" \
  echo "GITHUB_REPOSITORY=$GITHUB_REPOSITORY" \
  echo "BUILD_TIME=$BUILD_TIME"
  shell: /bin/bash -e {0}
GITHUB_REF=refs/pull/594/merge echo COMMIT_AUTHOR=Sebastian Titakis (1012854+stitakis@users.noreply.github.com) echo COMMIT_MESSAGE=Merge 974c41d694cd49c58f0b94d92993a0ce7975ce53 into 65ff5968e6e5606b3b2447dd5e64efd2cf0249ea echo COMMIT_SHA=1ffb4928dbf68504c069e52e20e0236febfdd8ea echo GITHUB_SHA=1ffb4928dbf68504c069e52e20e0236febfdd8ea echo COMMIT_TIME=2020-09-09 09:28:35 +0000 echo GITHUB_REPOSITORY=opendevstack/ods-provisioning-app echo BUILD_TIME=2020-09-09 09:28:51 +0000
```

**Comment by stitakis** [comment]
Fix worked in build of #592 -> branch 3.x is fixed.
